### PR TITLE
Orgs billing links

### PR DESCRIPTION
**Pull request by raptisj**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description
Edit billing options and member limit widget

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

### CODE REVIEW

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherPopover.tsx [hunk 1]
```diff
@@ -92,6 +110,7 @@ export const OrganizationSwitcherPopover = React.forwardRef<HTMLDivElement, Orga
                   sx={theme => ({ padding: `0 ${theme.space.$6}` })}
                 />
                 {manageOrganizationButton}
+                {__unstable_manageBillingUrl && billingOrganizationButton}
```
Comment (SokratisVidros) [inline packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherPopover.tsx:113]: 💯

File: packages/clerk-js/src/ui/components/OrganizationProfile/BillingWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        sx={t => ({ marginTop: t.space.$1 })}
+      />
+      <Text
+        variant='regularRegular'
+        sx={t => ({
+          fontWeight: t.fontWeights.$normal,
+          lineHeight: t.lineHeights.$tall,
+        })}
+      >
+        The invitations could not be sent. Fix the following and try again: <br />
```
Comment (SokratisVidros) [inline packages/clerk-js/src/ui/components/OrganizationProfile/BillingWidget.tsx:23]: @raptisj Didn't we change this copywriting? Please double-check with @alex-ntousias

**Review by SokratisVidros — APPROVED** [review]

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherPopover.tsx [hunk 3]
```diff
@@ -24,8 +25,12 @@ export const OrganizationSwitcherPopover = React.forwardRef<HTMLDivElement, Orga
     const { openOrganizationProfile, openCreateOrganization } = useCoreClerk();
     const { organization: currentOrg } = useCoreOrganization();
     const { isLoaded, setActive } = useCoreOrganizationList();
+    const router = useRouter();
     const {
       hidePersonal,
+      __unstable_manageBillingUrl,
```
Comment (SokratisVidros) [inline packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherPopover.tsx:31]: @nikosdouvlis What's the latest approach for `__unstable` props? Do we need to update `clerk.ts` in the `@clerk/types` package as well?

File: packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx [hunk 4]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 
 import { buildURL } from '../../utils/url';
 import { buildAuthQueryString, parseAuthProp } from '../common/authPropHelpers';
-import { useEnvironment } from '../contexts';
+import { useCoreClerk, useEnvironment } from '../contexts';
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx:5]: Why is this change needed?

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx [hunk 5]
```diff
@@ -27,12 +27,13 @@ export const OrganizationSwitcherTrigger = React.forwardRef<HTMLButtonElement, O
             gap={3}
             size={'sm'}
             organization={organization}
-            sx={{ maxWidth: '30ch' }}
+            sx={{ maxWidth: '30ch', minWidth: '190px' }}
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx:30]: Can you elaborate? This midwidth here looks unrelate to the billing feature

File: packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx (hunk 4 above)
Comment (raptisj) [reply packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx:5]: @nikosdouvlis sorry I'll remove

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx (hunk 5 above)
Comment (raptisj) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx:30]: @nikosdouvlis we need a min-width in case we have an org name with one letter.

File: packages/clerk-js/src/ui/components/OrganizationProfile/MembershipWidget.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        <Text
+          variant='regularRegular'
+          sx={t => ({
+            fontWeight: t.fontWeights.$normal,
+            lineHeight: t.lineHeights.$tall,
+            color: t.colors.$blackAlpha600,
+          })}
+        >
+          {organization?.membersCount + organization?.pendingInvitationsCount} of{' '}
+          {!__unstable_manageBillingMembersLimit ? 'unlimited' : __unstable_manageBillingMembersLimit} members
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/components/OrganizationProfile/MembershipWidget.tsx:58]: I think we don't need the optional chaining here.

File: packages/clerk-js/src/ui/components/OrganizationProfile/MembershipWidget.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    __unstable_manageBillingMembersLimit <= organization?.pendingInvitationsCount + organization?.membersCount;
+
+  return (
+    <Flex
+      sx={theme => ({
+        background: theme.colors.$blackAlpha50,
+        padding: theme.space.$4,
+        justifyContent: 'space-between',
+        alignItems: 'center',
+      })}
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/components/OrganizationProfile/MembershipWidget.tsx:25]: Let's make this as circular as the rows of the table?!

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx [hunk 8]
```diff
@@ -14,11 +14,13 @@ const _OrganizationSwitcher = withFloatingTree(() => {
...
-        onClick={toggle}
-        isOpen={isOpen}
-      />
+      <Flex>
+        <OrganizationSwitcherTrigger
+          ref={reference}
+          onClick={toggle}
+          isOpen={isOpen}
+        />
+      </Flex>
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx:23]: Why was this container added?

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherPopover.tsx (hunk 3 above)
Comment (nikosdouvlis) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherPopover.tsx:31]: We shouldn't type these anywhere as we don't want these props to visible at all - they're just going to get propagated as simple runtime values

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx [hunk 9]
```diff
@@ -14,11 +14,13 @@ const _OrganizationSwitcher = withFloatingTree(() => {
 
   return (
     <Flow.Root flow='organizationSwitcher'>
-      <OrganizationSwitcherTrigger
-        ref={reference}
-        onClick={toggle}
-        isOpen={isOpen}
-      />
+      <Flex>
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx:17]: Why is this change needed?

File: packages/clerk-js/src/ui/components/OrganizationProfile/MembershipWidget.tsx [hunk 10]
```diff
@@ -0,0 +1,63 @@
+import { useRouter } from '../../../ui/router';
+import { useCoreOrganization, useOrganizationProfileContext } from '../../contexts';
+import { Col, Flex, Link, Text } from '../../customizables';
+
+export const MembershipWidget = () => {
+  const { organization } = useCoreOrganization();
+  const { __unstable_manageBillingUrl, __unstable_manageBillingMembersLimit } = useOrganizationProfileContext();
+  const router = useRouter();
+
+  if (!organization) {
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/components/OrganizationProfile/MembershipWidget.tsx:10]: Would it be a nice idea to only show this if `__unstable_manageBillingUrl, __unstable_manageBillingMembersLimit` are defined as an extra safety check?

File: packages/clerk-js/src/ui/components/OrganizationProfile/BillingWidget.tsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+        <br />
+        <Link
+          variant='regularRegular'
+          sx={t => ({
+            alignSelf: 'flex-start',
+            color: t.colors.$primary500,
+            fontWeight: t.fontWeights.$normal,
+            ':hover': {
+              cursor: 'pointer',
+            },
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/components/OrganizationProfile/BillingWidget.tsx:33]: Not sure we need this selector

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx (hunk 8 above)
Comment (raptisj) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx:23]: @desiprisg there was an alignment issue. The box where the switcher was contained resembled a `flex-start` situation. But now I see that this is not longer the case. Something outside of it was probably causing the issue. I removed it.

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx (hunk 9 above)
Comment (raptisj) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx:17]: @nikosdouvlis https://github.com/clerkinc/javascript/pull/550#discussion_r1032393958

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx (hunk 8 above)
Comment (raptisj) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx]: @desiprisg @nikosdouvlis actually because of this. You will notice it is not align horizontally 
Screen Shot 2022-11-25 at 2 41 46 PM

File: packages/clerk-js/src/ui/components/OrganizationProfile/BillingWidget.tsx [hunk 12]
```diff
@@ -0,0 +1,41 @@
+import { useRouter } from '../../../ui/router';
+import { useOrganizationProfileContext } from '../../contexts';
+import { AlertIcon, Flex, Link, Text } from '../../customizables';
+
+export const BillingWidget = () => {
+  const { __unstable_manageBillingUrl, __unstable_manageBillingMembersLimit } = useOrganizationProfileContext();
+  const router = useRouter();
```
Comment (desiprisg) [inline packages/clerk-js/src/ui/components/OrganizationProfile/BillingWidget.tsx:7]: Should we turn these into props for the BillingWidget to force the outside check for them being defined?

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx (hunk 5 above)
Comment (desiprisg) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx:30]: I don't really understand. What occurs when there's an org with one letter?

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx (hunk 8 above)
Comment (nikosdouvlis) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcher.tsx:23]: Its not clear to me if this is a clerk-js or a dashboard issue

File: packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx (hunk 5 above)
Comment (raptisj) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx:30]: @nikosdouvlis @desiprisg this was a requirement from @SokratisVidros but since it interferes with the public lib I guess we should ditch it. Plus the breadcrumbs in the dashboard don't have a fixed width either.

Comment (SokratisVidros) [reply packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx]: @raptisj The fixed width of the <OrganizationSwitcher/> should be as styling applied from Dashboard, not added to the component for everyone. The component must be as flexible as possible. We have the appearance prop for that so that we can customize the styling when we use it.

**Review by desiprisg — APPROVED** [review]

**Review by nikosdouvlis — APPROVED** [review]
💯
